### PR TITLE
Support for "cumulative" plots

### PR DESCRIPTION
**Pull request by mhugo**
Adds support for "cumulative" plots, represented by bar diagrams, both for vertical and horizontal plots.

Capture d’écran de 2020-04-08 09-02-59
Capture d’écran de 2020-04-08 09-03-44
Capture d’écran de 2020-04-08 09-04-08

A bit of refactoring has also been made:
- merge of log_view.py and timeseries_view.py into plot_view.py

### CODE REVIEW

**Comment by mhugo** [comment]
cc @delhomer

File: data_selector.py [hunk 1]
```diff
@@ -122,40 +123,56 @@ def __load_feature(self, feature):
...
-                                       x_start_fieldname=cfg["start_measure_column"],
-                                       x_delta_fieldname=cfg["interval_column"])
+                    data = LayerData(
+                        data_l,
+                        cfg["event_column"],
+                        cfg["value_column"],
+                        filter_expression=filter_expr,
+                        uom=uom
+                    )
+                elif cfg["type"] == "cumulative":
```
Comment (delhomer) [inline data_selector.py]: What about the "continuous" type? Is it definitely obsolete? (We might think about diagraphic data, which are still continuous in our current use case...)

**Review by delhomer — APPROVED** [review]
Huge work, thank you @mhugo !

File: data_selector.py (hunk 1 above)
Comment (mhugo) [reply data_selector.py]: Thanks ! That should be fixed now

File: data_selector.py [hunk 2]
```diff
@@ -122,40 +125,78 @@ def __load_feature(self, feature):
...
+                    )
+                    self.__viewer.add_data_cell(
+                        data,
+                        title,
+                        uom,
+                        station_name=feature_name,
+                        config=cfg
+                    )
+                elif cfg["type"] == "cumulative":
+                    data = IntervalData(
```
Comment (delhomer) [inline data_selector.py:177]: Sorry to go back to this merged feature... I'm looking at the source code to understand how it does work for me.

Not sure to understand this part of the code: here, is the `data` variable used in any way?

File: data_selector.py [hunk 3]
```diff
@@ -122,40 +125,78 @@ def __load_feature(self, feature):
...
+                        data_l,
+                        cfg["min_event_column"],
+                        cfg["max_event_column"],
+                        cfg["value_column"],
+                        filter_expression=filter_expr,
+                        uom=uom
+                    )
+                    self.__viewer.add_histogram(
+                        data_l,
```
Comment (delhomer) [inline data_selector.py:186]: In line with the previous comment, maybe we should replace `data_l` with `data.get_layer()`, as for the `instantaneous` case?

File: data_selector.py (hunk 2 above)
Comment (mhugo) [reply data_selector.py:177]: Thanks for the review. IntervalData is actually not used anymore

Comment (mhugo) [reply data_selector.py:177]: Fixed in https://github.com/Oslandia/QGeoloGIS/commit/37cee35695dddc9fad15d81bbcfc544680435222

File: data_selector.py (hunk 3 above)
Comment (mhugo) [reply data_selector.py:186]: See above

**Pull request merged**